dask/base.py

```diff
@@ -211,7 +211,24 @@ def is_dask_collection(x) -> bool:
 
     """
     try:
-        return x.__dask_graph__() is not None
+        if "xarray" in type(x).__module__:
+            import xarray
+
+            if isinstance(x, xarray.Dataset):
+                return any(is_dask_collection(v._data) for _, v in x.variables.items())
+            elif isinstance(x, xarray.DataArray):
+                return is_dask_collection(x.variable._data)
```

Comment:
This is returning a false negative in the (admittedly niche) event you have an eager variable and lazy non-index coordinates

```python
>>> import dask.array as da, xarray
>>> xarray.DataArray([1,2], dims=["x"], coords={"x": [10, 20], "x2": ("x", da.zeros(2))})
<xarray.DataArray (x: 2)>
array([1, 2])
Coordinates:
  * x        (x) int64 10 20
    x2       (x) float64 dask.array<chunksize=(2,), meta=np.ndarray>
```